pkg/globalmanager/incrementallearningjob.go

```diff
@@ -0,0 +1,794 @@
+	startTime := time.Now()
+	defer func() {
+		klog.V(4).Infof("Finished syncing incrementallearning job %q (%v)", key, time.Since(startTime))
+	}()
+
+	ns, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return false, err
+	}
+	if len(ns) == 0 || len(name) == 0 {
+		return false, fmt.Errorf("invalid incrementallearning job key %q: either namespace or name is missing", key)
+	}
+	sharedIncrementalJob, err := jc.jobLister.IncrementalLearningJobs(ns).Get(name)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			klog.V(4).Infof("incrementallearning job has been deleted: %v", key)
+			return true, nil
+		}
+		return false, err
+	}
+	incrementaljob := *sharedIncrementalJob
+	// set kind for incrementaljob in case that the kind is None
+	incrementaljob.SetGroupVersionKind(neptunev1.SchemeGroupVersion.WithKind("IncrementalLearningJob"))
+	// incrementaljob first start, create pod for inference
+	if incrementaljob.Status.StartTime == nil {
+		now := metav1.Now()
+		incrementaljob.Status.StartTime = &now
+		pod := jc.getSpecifiedPods(&incrementaljob, "inference")
+		if pod == nil {
+			err = jc.createInferPod(&incrementaljob)
+		} else {
+			if pod.Status.Phase != v1.PodRunning && pod.Status.Phase != v1.PodPending {
+				err = jc.createInferPod(&incrementaljob)
+			}
+		}
+		if err != nil {
+			return false, nil
+		}
+	}
+
+	// if incrementaljob was finished previously, we don't want to redo the termination
+	if IsIncrementalJobFinished(&incrementaljob) {
+		return true, nil
+	}
+
+	forget := false
+	jobFailed := false
+	needUpdated := false
+
+	// update conditions of incremental job
+	needUpdated, err = jc.updateIncrementalJobConditions(&incrementaljob)
+
+	if needUpdated {
+		if err := jc.updateIncrementalJobStatus(&incrementaljob); err != nil {
+			return forget, err
+		}
+
+		if jobFailed && !IsIncrementalJobFinished(&incrementaljob) {
+			// returning an error will re-enqueue IncrementalJob after the backoff period
+			return forget, fmt.Errorf("failed pod(s) detected for incrementaljob key %q", key)
+		}
+
+		forget = true
+	}
+
+	return forget, err
+}
+
+// updateIncrementalJobConditions ensures that conditions of incrementallearning job can be changed by podstatus
+func (jc *IncrementalJobController) updateIncrementalJobConditions(incrementaljob *neptunev1.IncrementalLearningJob) (bool, error) {
+	var initialType neptunev1.ILJobStageConditionType
+	var latestCondition neptunev1.ILJobCondition = neptunev1.ILJobCondition{
+		Stage: neptunev1.ILJobTrain,
+		Type:  initialType,
+	}
+	var newConditionType neptunev1.ILJobStageConditionType
+	latestCondition.Stage = neptunev1.ILJobTrain
+	var needUpdated = false
+	jobConditions := incrementaljob.Status.Conditions
+	var podStatus v1.PodPhase = v1.PodUnknown
+	if len(jobConditions) > 0 {
+		// get latest pod and pod status
+		latestCondition = (jobConditions)[len(jobConditions)-1]
+		klog.V(2).Infof("latest stage %v:", latestCondition.Stage)
```

Comment:
fill this log with job info, otherwise we don't know which job.

Reply:
ok